core/common/src/main/scala/org/typelevel/otel4s/context/Context.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s
+package context
+
+trait Context[C] { outer =>
```

Comment:
Since we treat this interface as a typeclass, should we rename it to `ContextOps`? And `ContextOps` (line 44) to `ContextSyntax`?

Reply:
Or, perhaps, `Contextual`.

Reply:
Either option works for me. 

Once we have real users of a new typeclass yet (e.g. some middleware), we can reconsider our choices and improve the UX :)